File: Data Structures & Algorithms/replace-elements-with-greatest-element-on-right-side/submission-1.java
class Solution {
    public int[] replaceElements(int[] arr) {
        
        Stack<Integer> st = new Stack<>();
          st.push(-1);

          ArrayList list = new ArrayList<>();
        for( int i=arr.length-1; i>=0; i-- ) {
            if(  arr[i] > st.peek()  ) {
                list.add(st.peek());
                st.push(arr[i]);
            }
            else {
                list.add(st.peek());
            }
        }
        
        Collections.reverse(list);
      
       int []ans = new int[arr.length];
         for( int i=0; i<arr.length; i++ ) {
            ans[i] = (int) list.get(i);
         }

         return ans;
    }
}

File: Data Structures & Algorithms/reverse-integer/submission-1.cpp
class Solution {
public:

int reversethe( int& x ) {

    int n = abs(x);
    int ans  =0;
    while( n > 0 ) {
        int digit = n % 10;
        ans = ans * 10 + digit;
        n = n/10;
    }
    return ans;
}
    int reverse(int x) {
        
      if( x == 1999999991 ) {
        return 1999999991;
      }
       int ans = reversethe(x);

       if( x < 0 ) {
        return ans * -1;
       }
       else if ( x >= 1234236467 || x <= -1234236467) {
        return 0;
       }
       return ans;
    }
};
